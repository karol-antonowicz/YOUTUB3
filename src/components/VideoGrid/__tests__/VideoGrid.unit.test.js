import React from 'react';
import {shallow} from 'enzyme';
import {VideoGrid} from '../VideoGrid';

describe('VideoGrid', ()=> {
    test('renders VideoGrid without props', () => {
        const wrapper = shallow(<VideoGrid/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders with props title empty string', ()=> {
        const wrapper = shallow(<VideoGrid title=''/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('render VideoGrid with no title props and hideDivider', ()=> {
        const wrapper = shallow(<VideoGrid hideDivider={true}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders with title and hideDivider props set to true', () => {
        const wrapper = shallow(<VideoGrid title='Black & White' hideDivider={true}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders with title and hideDivider props set to false', () => {
        const wrapper = shallow(<VideoGrid title='Black & White' hideDivider={false}/>);
        expect(wrapper).toMatchSnapshot();
    });

})