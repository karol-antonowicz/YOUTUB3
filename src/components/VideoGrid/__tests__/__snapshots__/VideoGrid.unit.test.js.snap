// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid render VideoGrid with no title props and hideDivider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid without props 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders with props title empty string 1`] = `
<Fragment>
  <VideoGridHeader
    title=""
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders with title and hideDivider props set to false 1`] = `
<Fragment>
  <VideoGridHeader
    title="Black & White"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders with title and hideDivider props set to true 1`] = `
<Fragment>
  <VideoGridHeader
    title="Black & White"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;
